### ISSUE TITLE
SLANT option makes Long files unreadable

### ISSUE THREAD

**Issue by sultamehr** [cta-observatory/pycorsikaio#40]
Hello, 

This might be a user issue, rather than an issue with the code. I am hoping I can get some feed back. 

 I am using CORSIKA v 77500. With the Slant option on, I get the following error 
![Screenshot 2024-02-16 at 17 29 24](https://github.com/cta-observatory/pycorsikaio/assets/49162909/30816c86-9fac-4b73-b7a4-8928a504c307)

"ValueError: Some errors were detected !
    Line #679 (got 9 columns instead of 10)"

Is there something required for the function "read_longitudinal_distributions(path)" for the slant option?

Thank you so much. Sorry for a random question. I am new to CORSIKA and found your package helpful to learn.

Best,

Mehreen

**Comment by maxnoe** [comment]
Dear Sultamehr,

Could you provide a test file, inputcard and compilation options that exhibit this issue?

**Comment by maxnoe** [comment]
There is a unit test, testing reading of longitudinal files with the SLANT option here:

https://github.com/cta-observatory/pycorsikaio/blob/main/tests/test_longitudinal.py

this is the test file used in that test:

https://github.com/cta-observatory/pycorsikaio/blob/main/tests/test_longitudinal.py

**Comment by sultamehr** [comment]
Hello! 

Thank you for responding. Here is the data card:
```
RUNNR   757                              run number
EVTNR   1                              number of first shower event
NSHOW   1                              number of showers to generate
PRMPAR  14                             particle type of prim. particle
ESLOPE  -2.7                           slope of primary energy spectrum
ERANGE  1.E8  1.E8                     energy range of primary particle
THETAP  0.  45.                       range of zenith angle (degree)
PHIP    -180.  180.                    range of azimuth angle (degree)
SEED    1   0   0                      seed for 1. random number sequence
SEED    2   0   0                      seed for 2. random number sequence
OBSLEV  5.E5                           observation level (in cm)
MAGNET  20.0  42.8                     magnetic field centr. Europe
HADFLG  0  0  0  0  0  2               flags hadr.interact.&fragmentation
ECUTS   .3  0.3  1e-4  1e-4         energy cuts for particles
MUADDI  F                              additional info for muons
MUMULT  F                              muon multiple scattering angle
ELMFLG  F   T                          em. interaction flags (NKG,EGS)
RADNKG  200.E2                         outer radius for NKG lat.dens.distr.
LONGI   T  1.  F  T                   longit.distr. & step size & fit & out
MAXPRT  100                              max. number of printed events
DIRECT  ./dataoutput/                             output directory
USER    you                            user
DEBUG   F  6  F  1000000               debug flag and log.unit for out
EXIT                                   terminates input
```

I had QGSII + GHEISHA with CERENKOV, CURVED, SLANT options.  (attaching the output text file for reference).

[output_QGSII_gheisha_curved_SLANT_test_with_cerenkov.txt](https://github.com/cta-observatory/pycorsikaio/files/14349257/output_QGSII_gheisha_curved_SLANT_test_with_cerenkov.txt)

[DATA000757_LONG.zip](https://github.com/cta-observatory/pycorsikaio/files/14349234/DATA000757_LONG.zip)

Any help you can provide would be great! I will try to see if I can run the test from the package to compare with.

**Comment by fschlueter** [comment]
Hallo, stumbled across this issue on which I might can shed some light. I might have had the same error, for me the issue is that an entry in the energy deposit table is negative. In that case no space as delimiter is between columns, e.g., `678.5 1.93174E+05-2.13648E+04 1.02131E+05`. The same can be found in the example file shared by @sultamehr. You can regex those cases with `E[+-]\d{2}-\d`.

**Comment by maxnoe** [comment]
This seems like a bug in CORSIKA and not here in the reader.

Does this only happen with `GHEISHA`?

**Comment by fschlueter** [comment]
I agree, negative energy deposits are unphysical. I found it in simulations made with UrQMD.

**Comment by fschlueter** [comment]
In the situations I found, the energy deposit was only barely negative, so I imagine that being ~ irrelevant.

**Comment by maxnoe** [comment]
The CORSIKA maintainers confirmed this bug existed, but said it should be fixed in current versions of CORSIKA.

Just to be safe: are you seeing this with the current CORSIKA 7 release? Or only with older versions?

**Comment by cxwx** [comment]
same issue, though the probability is extremely low
such phenomena are quite common in numerical computing; they can typically be managed by implementing proper conditional checks
But then again, is it really worth adding extra logic for such rare outliers.

**Comment by fschlueter** [comment]
nope with some older version (7.74 or something similar if I recall correctly)
